install/infra/terraform/gke/main.tf

```diff
@@ -0,0 +1,191 @@
+terraform {
+  required_version = ">= 1.0.3"
+}
+
+terraform {
+  backend "gcs" {
+    bucket = "gitpod-gke"
+    prefix = "tf-state"
+  }
+}
+
+provider "google" {
+  project     = var.project
+  credentials = var.credentials
+  region      = var.region
+  zone        = var.zone
+}
+
+resource "google_compute_network" "vpc" {
+  name                    = "vpc-${var.name}"
+  auto_create_subnetworks = "false"
+}
+
+resource "google_compute_subnetwork" "subnet" {
+  name          = "subnet-${var.name}"
+  region        = var.region
+  network       = google_compute_network.vpc.name
+  ip_cidr_range = "10.255.0.0/16"
+
+  secondary_ip_range {
+    range_name    = "cluster-secondary-ip-range"
+    ip_cidr_range = "10.0.0.0/12"
+  }
+
+  secondary_ip_range {
+    range_name    = "services-secondary-ip-range"
+    ip_cidr_range = "10.64.0.0/12"
+  }
+}
+
+resource "google_container_cluster" "gitpod-cluster" {
```

Comment:
Noting this for information sharing, no action needed.

This module handles a number of cases that are great for a production cluster (and thus would be great for a reference architecture module), but may be overkill for testing. Features like cluster autoscaling or regional clusters will provide better performance and resilience for production traffic, but for everyday system testing we shouldn't need a lot of nodes, autoscaling, or resilience in the case of a failed GCP zone.

Having a single, unified module for system tests (automated and manual) as well as reference architectures will be fantastic, and this will be great for that - so I don't think we need to pull out any features. However keeping an eye towards the overall cost of test clusters gives us an advantage; if we can drive down the cost of a single test run then we can run tests more cheaply and more frequently.